### ISSUE TITLE
Bug in Stopping Server

### ISSUE THREAD

**Issue by Anzide** [Minestom/Minestom#1636]
![image](https://user-images.githubusercontent.com/37073164/209917441-51940cc9-870d-414e-b219-e007f3c9e9ab.png)
My extension save players' data on PlayerDissconnentEvent. And PlayerDissconnentEvent will not be triggered if the server stop immediately after kicking all the players, just like the code above. That's because PlayerConnection.disconnect() method remove the player and dispatch PlayerDissconnentEvent in next tick, not immediately.

**Comment by Timofffee** [comment]
I don't think it looks like a mistake. The server can delay the player's hit to prevent the main loop from slowing down while waiting for the player to disconnect. Well, or is it due to the fact that the player is disabled in tick n. After that, the server should finish the tick and send all the packets. Then it seems quite logical that the server considers the player disabled only in the next tick.

**Comment by TheMeinerLP** [comment]
Can you add some code of reproduce ?
